### PR TITLE
Feature/rest server

### LINKED ISSUE #66: Add a flask endpoint in server.py
`GET http://example.com/api/<guild_id>/memes` list memes
`GET http://example.com/api/<guild_id>/memes/<memename>`  meme json
`GET http://example.com/api/<guild_id>/memes/<memename>/render?text=<text>&text=<text>`  render meme (returns an image)
`POST http://example.com/api/<guild_id>/memes/?token=<nonce>`  save meme json

### PR DESCRIPTION
**Pull request by ironchefpython**
closes #66

### CODE REVIEW

**Comment by ironchefpython** [comment]
Test by running `python3.7 -m nachomemes.server`
And then navigate to `http://127.0.0.1:5000/default/123/memes/nacho/render?text=Nacho&text=Meme`

File: nachomemes/bot.py [hunk 1]
```diff
@@ -347,27 +348,14 @@ def _find_close_matches(name, guild: Guild):
         "--local", action="store_true", help="Force running without Dynamo."
     )
     args = parser.parse_args()
+
     global testing
-    testing = args.debug
+    testing = args.debug
```
Comment (jswny) [inline nachomemes/bot.py]: Yup, this is a Chris PR alright.

File: nachomemes/store.py [hunk 2]
```diff
@@ -149,20 +150,41 @@ def render(self, message: Iterable[str], output: IO):
...
 
     @abstractmethod
-    def read_meme(
+    def get_meme(
         self, guild: Optional[Guild], id: str, increment_use: bool = False
-    ) -> MemeTemplate:
+    ) -> dict:
         pass
 
+    def meme(
```
Comment (jswny) [inline nachomemes/store.py]: What's the difference between `meme` and `get_meme`?

Comment (ironchefpython) [reply nachomemes/store.py]: I pointlessly renamed it

Comment (jswny) [reply nachomemes/store.py]: But `meme` and `get_meme` both exist?

Also, should this be `@abstractmethod`?

**Comment by jswny** [comment]
For reference:
- `read_meme` -> `get_template`
- `get_template_data` returns a `Template` as a `Dict`

**Pull request merged**